frontend/components/panels/VaultLiquidationLimitsCheckPanel.vue

```diff
@@ -0,0 +1,119 @@
+            <span v-if="liquidationLimits"
+                ><Explain :text="liquidationLimits.maximumCollateralDebtDai.toFixed()"></Explain> -
+                <Explain :text="liquidationLimits.currentCollateralDebtDai.toFixed()"></Explain> -
+                <Explain :text="debtDai.toFixed()"></Explain> =
+                <FormatCurrency :value="collateralDifference" currency="DAI"
+            /></span>
+            <div v-else>
+                <span class="opacity-50">Unknown</span>
+                <span>DAI</span>
+            </div>
+        </div>
+        <div class="flex justify-end mt-4">
+            <BaseButton
+                type="primary"
+                :disabled="isRefreshing"
+                :is-loading="isRefreshing"
+                @click="$emit('refreshLimits')"
+            >
+                Refresh
+            </BaseButton>
+        </div>
+    </BasePanel>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import BaseButton from '~/components/common/inputs/BaseButton.vue';
+import TextBlock from '~/components/common/other/TextBlock.vue';
+import Explain from '~/components/common/other/Explain.vue';
+import FormatCurrency from '~/components/common/formatters/FormatCurrency.vue';
+import BasePanel from '~/components/common/other/BasePanel.vue';
+import { LiquidationLimits } from '~/../core/src/types';
+
+export default Vue.extend({
+    name: 'VaultLiquidationLimitsCheckPanel',
+    components: {
+        BaseButton,
+        BasePanel,
+        TextBlock,
+        Explain,
+        FormatCurrency,
+    },
+    props: {
+        liquidationLimits: {
+            type: Object as Vue.PropType<LiquidationLimits>,
+            required: true,
+        },
+        debtDai: {
+            type: BigNumber,
+            required: true,
+        },
+        collateralType: {
+            type: String,
+            required: true,
+        },
+        isRefreshing: {
+            type: Boolean,
+            default: false,
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+    },
+    computed: {
+        globalDifference(): BigNumber {
+            return this.liquidationLimits.maximumProtocolDebtDai
+                .minus(this.liquidationLimits.currentProtocolDebtDai)
+                .minus(this.debtDai);
+        },
+        collateralDifference(): BigNumber {
+            return this.liquidationLimits.maximumCollateralDebtDai
+                .minus(this.liquidationLimits.currentCollateralDebtDai)
+                .minus(this.debtDai);
+        },
+        isGlobalLimitsReached(): Boolean {
+            if (this.globalDifference.isNegative()) {
+                return true;
+            }
+            if (this.collateralDifference.isNegative()) {
+                return true;
+            }
+            return false;
+        },
+        currentStateAndTitle(): PanelProps {
```

Comment:
You also need to add a state for when no `LiqudationLimits` are provided. Unless...

We decide to let the user continue with the transaction and MetaMask will inform them when they try to execute. 

What do you think @valiafetisov?

Reply:
I think they have to be provided according to the types (so that vault without them shouldn't exist, unless it's a liquidated vault). But we can still have `inactive state` for this case just for the stories

Reply:
Then let's make the prop `liquidationLimits` optional, but keep the types.ts intact